### ISSUE TITLE
Unable to use pyload for torrents through flexget

### ISSUE THREAD

**Issue by kepi** [pyload/pyload#3853]
### Description

I have kind of working *flexget*/*pyload* setup with RSS feeds and classic
hosters with help of *RealDebrid* account.

What I wanted to make it better is to use discover with torrents or some
torrents RSS feeds and then use *RealDebrid's* torrent feature to download it
through *pyload*.

It seems that *flexget* is trying to use `parseURLs` endpoint through `html`
attribute to validate links that are ok but *pyload* can't handle `magnet` ones.

I'm not sure if it is better handled on *flexget's* side or *pyload's*, but it seems to me
that as *pyload* can handle `magnet` links through UI, it should be able to do
through API too.

If you need additional details, let me know.

### Debug log

Using latest `stable` branch and latest `flexget`.

```
2020-12-07 18:34:32 DEBUG    pyload        discover-series-test Parsing url "magnet:?xt=urn:btih:xxxx.....%2Fannounce"
2020-12-07 18:34:32 DEBUG    utils.requests discover-series-test POSTing URL http://localhost:8000/api/parseURLs with args () and kwargs {'data': {'html': '" magnet%3A%3Fxt%3Durn%3Abtih%3Axxxx....announce"', 'url': '"magnet:?xt=urn:btih:xxxxxx......announce"', 'session': 'abcdefgh'}, 'json': None, 'timeout': 30}
2020-12-07 18:34:32 CRITICAL task          discover-series-test BUG: Unhandled error in plugin pyload: 500 Server Error: Internal Server Error for url: http://localhost:8000/api/parseURLs requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://localhost:8000/api/parseURLs
2020-12-07 18:34:32 CRITICAL manager       discover-series-test An unexpected crash has occurred. Writing crash report to /home/filmiky/.config/flexget/crash_report.2020.12.07.183432459168.log. Please verify you are running the latest version of flexget by using "flexget -V" from CLI or by using version_checker plugin at http://flexget.com/wiki/Plugins/version _checker. You are currently using version 3.1.33
2020-12-07 18:34:32 DEBUG    manager       discover-series-test Traceback: requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://localhost:8000/api/parseURLs
2020-12-07 18:34:32 WARNING  task          discover-series-test Aborting task (plugin: pyload)
```

This can be easily reproduced with using same approach through API manually, i.e.:

```python
import requests
payload_login = {'username': 'flexget', 'password': 'superpassword....'}
rs = requests.post('http://localhost:8000/api/login', data=payload_login, json=None, timeout=30 )
session_id = rs.json
payload = {'html': '" magnet%3A%3Fxt%3Durn%3Abtih%3Axxxxxx..........announce"', 'session': session_id}
rs = requests.post('http://localhost:8000/api/parseURLs', data=payload, json=None, timeout=30)
```

**Comment by GammaC0de** [comment]
Please test

**Comment by kepi** [comment]
Thanks for such fast response! Have to leave for today, but at least quick test and failing on host part:


```
Traceback (most recent call last):
File "/home/filmiky/apps/pyload/module/web/api_app.py", line 70, in call_api
return callApi(func, *args, **kwargs)
File "/home/filmiky/apps/pyload/module/web/api_app.py", line 84, in callApi
**dict([(x, literal_eval(y)) for x, y in kwargs.iteritems()]))
File "/home/filmiky/apps/pyload/module/Api.py", line 351, in parseURLs
page = getURL(url)
File "/home/filmiky/apps/pyload/module/network/RequestFactory.py", line 131, in getURL
return pyreq.getURL(*args, **kwargs)
File "/home/filmiky/apps/pyload/module/network/RequestFactory.py", line 72, in getURL
rep = h.load(*args, **kwargs)
File "/home/filmiky/apps/pyload/module/network/HTTPRequest.py", line 256, in load
self.c.perform()
error: (6, 'Could not resolve host: magnet')
/home/filmiky/apps/pyload/module/web/api_app.py:75: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
return json.dumps({"error": e.message, "traceback": format_exc()})
```

for completeness flexget error:


```
2020-12-08 01:29:39 VERBOSE  details       discover-series-test Summary - Accepted: 1 (Rejected: 0 Undecided: 8 Failed: 0)
2020-12-08 01:29:39 DEBUG    utils.requests discover-series-test POSTing URL http://localhost:8000/api/login with args () and kwargs {'data': {'username': 'flexget', 'password': '....'}, 'json': None, 'timeout': 30}
2020-12-08 01:29:39 DEBUG    pyload        discover-series-test Parsing url "magnet:?xt=urn:btih:.....announce"
2020-12-08 01:29:39 DEBUG    utils.requests discover-series-test POSTing URL http://localhost:8000/api/parseURLs with args () and kwargs {'data': {'html': '" magnet%3A%3Fxt%3Durn%3Abtih:.....announce"', 'session': '....'}, 'json': None, 'timeout': 30}
2020-12-08 01:29:39 CRITICAL task          discover-series-test BUG: Unhandled error in plugin pyload: 500 Server Error: Internal Server Error for url: http://localhost:8000/api/parseURLs
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 885, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x7f8658516400>
    └ <Thread(task_queue, started daemon 140214873368320)>
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x7f86585161e0>
    └ <Thread(task_queue, started daemon 140214873368320)>
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(task_queue, started daemon 140214873368320)>
    │    │        │    └ ()
    │    │        └ <Thread(task_queue, started daemon 140214873368320)>
    │    └ <bound method TaskQueue.run of <flexget.task_queue.TaskQueue object at 0x7f86528ffeb8>>
    └ <Thread(task_queue, started daemon 140214873368320)>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/task_queue.py", line 46, in run
    self.current_task.execute()
    │    │            └ <function Task.execute at 0x7f8655a8f0d0>
    │    └ <flexget.task.Task object at 0x7f861ecd7668>
    └ <flexget.task_queue.TaskQueue object at 0x7f86528ffeb8>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/task.py", line 80, in wrapper
    return func(self, *args, **kw)
           │    │      │       └ {}
           │    │      └ ()
           │    └ <flexget.task.Task object at 0x7f861ecd7668>
           └ <function Task.execute at 0x7f8655a8f048>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/task.py", line 697, in execute
    self._execute()
    │    └ <function Task._execute at 0x7f8655a8df28>
    └ <flexget.task.Task object at 0x7f861ecd7668>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/task.py", line 663, in _execute
    self.__run_task_phase(phase)
    │                     └ 'output'
    └ <flexget.task.Task object at 0x7f861ecd7668>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/task.py", line 489, in __run_task_phase
    response = self.__run_plugin(plugin, phase, args)
               │                 │       │      └ (<flexget.task.Task object at 0x7f861ecd7668>, {'queue': False, 'folder': "/data/downloads/pyload/series/{{ trakt_series_name...
               │                 │       └ 'output'
               │                 └ <PluginInfo(name=pyload)>
               └ <flexget.task.Task object at 0x7f861ecd7668>
> File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/task.py", line 522, in __run_plugin
    result = method(*args, **kwargs)
             │       │       └ {}
             │       └ (<flexget.task.Task object at 0x7f861ecd7668>, {'queue': False, 'folder': "/data/downloads/pyload/series/{{ trakt_series_name...
             └ <Event(name=plugin.pyload.output,func=on_task_output,priority=128)>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/event.py", line 22, in __call__
    return self.func(*args, **kwargs)
           │    │     │       └ {}
           │    │     └ (<flexget.task.Task object at 0x7f861ecd7668>, {'queue': False, 'folder': "/data/downloads/pyload/series/{{ trakt_series_name...
           │    └ <bound method PluginPyLoad.on_task_output of <flexget.plugins.clients.pyload.PluginPyLoad object at 0x7f865295e5c0>>
           └ <Event(name=plugin.pyload.output,func=on_task_output,priority=128)>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/plugins/clients/pyload.py", line 121, in on_task_output
    self.add_entries(task, config)
    │    │           │     └ {'queue': False, 'folder': "/data/downloads/pyload/series/{{ trakt_series_name|default(series_name)|pathscrub|replace('&',' a...
    │    │           └ <flexget.task.Task object at 0x7f861ecd7668>
    │    └ <function PluginPyLoad.add_entries at 0x7f8653491ea0>
    └ <flexget.plugins.clients.pyload.PluginPyLoad object at 0x7f865295e5c0>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/plugins/clients/pyload.py", line 159, in add_entries
    result = api.post('parseURLs', data=data)
             │   │                      └ {'html': '" magnet%3A%3Fxt%3Durn%3A
             │   └ <function PyloadApi.post at 0x7f8653491d90>
             └ <flexget.plugins.clients.pyload.PyloadApi object at 0x7f861e963ba8>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/plugins/clients/pyload.py", line 42, in post
    return self.requests.post(self.url.rstrip("/") + "/" + method.strip("/"), data=data)
           │    │        │    │    │   │                   │      │                └ {'html': '" magnet%3A%3Fxt%3Durn%3A
           │    │        │    │    │   │                   │      └ <method 'strip' of 'str' objects>
           │    │        │    │    │   │                   └ 'parseURLs'
           │    │        │    │    │   └ <method 'rstrip' of 'str' objects>
           │    │        │    │    └ 'http://localhost:8000/api'
           │    │        │    └ <flexget.plugins.clients.pyload.PyloadApi object at 0x7f861e963ba8>
           │    │        └ <function Session.post at 0x7f865688bd08>
           │    └ <flexget.utils.requests.Session object at 0x7f86435b55f8>
           └ <flexget.plugins.clients.pyload.PyloadApi object at 0x7f861e963ba8>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/requests/sessions.py", line 581, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
           │    │               │         │          │       └ {}
           │    │               │         │          └ None
           │    │               │         └ {'html': '" magnet%3A%3Fxt%3Durn%3A
           │    │               └ 'http://localhost:8000/api/parseURLs'
           │    └ <function Session.request at 0x7f8655aec510>
           └ <flexget.utils.requests.Session object at 0x7f86435b55f8>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/utils/requests.py", line 260, in request
    result.raise_for_status()
    │      └ <function Response.raise_for_status at 0x7f865687b620>
    └ <Response [500]>
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
          │         │                        └ <Response [500]>
          │         └ '500 Server Error: Internal Server Error for url: http://localhost:8000/api/parseURLs'
          └ <class 'requests.exceptions.HTTPError'>

requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://localhost:8000/api/parseURLs
Traceback (most recent call last):
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/task.py", line 522, in __run_plugin
    result = method(*args, **kwargs)
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/event.py", line 22, in __call__
    return self.func(*args, **kwargs)
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/plugins/clients/pyload.py", line 121, in on_task_output
    self.add_entries(task, config)
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/plugins/clients/pyload.py", line 159, in add_entries
    result = api.post('parseURLs', data=data)
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/plugins/clients/pyload.py", line 42, in post
    return self.requests.post(self.url.rstrip("/") + "/" + method.strip("/"), data=data)
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/requests/sessions.py", line 581, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/flexget/utils/requests.py", line 260, in request
    result.raise_for_status()
  File "/home/filmiky/bin/installs/flexget/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://localhost:8000/api/parseURLs
```

**Comment by GammaC0de** [comment]
Since I don't have Flexget, can you please test this file before I commit it?
Please test normal URLS also, not just magnets.
[Api.zip](https://github.com/pyload/pyload/files/5662175/Api.zip)

**Comment by kepi** [comment]
Looks promising. I got messed up when trying your first fix because by accident I turned on `parse_url` option in *Flexget* couple minutes ago. This option probably put magnet link to `url` field too. I even wrote patch to work around this before I realized that this isn't how it should work :) Your first fix in commit:00ab9a5 was probably OK but it seems that version from attached .zip is working too.

Anyway I don't have exactly great setup for running tests, it will be best to test it "in production" :) I'll got back to you hopefully tomorrow with result for normal URLs.

**Comment by kepi** [comment]
Normal links are passing without any trouble. I believe torrent works too. If there would be any trouble I hit previously, I'll send PR with fix.

I wanted to test it more but RD had huge outage yesterday and I had no time today so far.

Thanks again, I really appreciate your quick help!
